src/object_store/src/object/opendal_engine/opendal_object_store.rs

```diff
@@ -59,6 +62,22 @@ impl OpendalObjectStore {
     }
 }
 
+pub fn new_http_client(config: &ObjectStoreConfig) -> ObjectResult<HttpClient> {
+    let mut client_builder = reqwest::ClientBuilder::new();
+
+    if let Some(keepalive_ms) = config.s3.object_store_keepalive_ms.as_ref() {
+        client_builder = client_builder.tcp_keepalive(Duration::from_millis(*keepalive_ms));
+    }
+
+    if let Some(nodelay) = config.s3.object_store_nodelay.as_ref() {
+        client_builder = client_builder.tcp_nodelay(*nodelay);
+    }
```

Comment:
It is strange to change the behavior non-s3 engine based on s3 configs. We should either move `keep_alive_ms` and `nodelay` configs out of s3 configs or only change the behavior of s3 engine based on the current configs.